backend/src/main/java/com/bakdata/conquery/apiv1/query/QueryDescription.java

```diff
@@ -111,4 +112,7 @@ public static void authorizeQuery(QueryDescription queryDescription, Subject sub
 		}
 	}
 
+	default Optional<Set<Integer>> collectRequiredEntities(){
+		return Optional.empty();
+	}
```

Comment:
Das über all die Methoden hinzugekommen sind, gefällt mir nicht so. 
Ich schlage vor, dass du ein Interface schreibst, dass diese Methode deklariert und dann mit dem Visitor-Pattern durch den Baumgehst.

Analog würde ich es jetzt auch für `collectRequiredQueries` andenken. Das würde uns einige Boilerplate methoden sparen.

Reply:
Ich bin mir nicht ganz sicher was du meinst, bzw wie ich das dann mit AND/OR machen würde, kannst du deine Änderung hier einfach reinpushen? Musst sie ja nicht ganz durchziehen

Reply:
Habe meinen Vorschlag hier gepush: https://github.com/bakdata/conquery/pull/2815/commits/1f77792909442550348a2a096a2a8c52d89c95b5